### PR TITLE
fixed report to handle a paper with multiple cgc IDs properly

### PR DESCRIPTION
**Pull request by sweng66**

### CODE REVIEW

File: agr_literature_service/lit_processing/data_ingest/dqm_ingest/sort_dqm_json_reference_updates.py [hunk 1]
```diff
@@ -204,6 +204,22 @@ def sort_dqm_references(input_path, input_mod, base_dir=base_path):      # noqa:
     XREF_to_resource_id = {x["curie"]: x["resource_id"] for x in db_session.execute(
         "SELECT curie, resource_id FROM cross_reference WHERE resource_id is not null "
         "AND is_obsolete is False").fetchall()}
+    rows = db_session.execute(
+        "SELECT r.curie, c.curie_prefix, c.curie FROM reference r, cross_reference c "
+        "WHERE c.curie_prefix = 'CGC' "
+        "AND c.is_obsolete is False "
+        "AND r.reference_id = c.reference_id").fetchall()
+    ref_cgcs_valid = {}
```
Comment (valearna) [inline agr_literature_service/lit_processing/data_ingest/dqm_ingest/sort_dqm_json_reference_updates.py:212]: You can simplify this part (lines 212-222) using a defaultdict:

```
ref_cgcs_valid = defaultdict(lambda: defaultdict(set))
for agr, prefix, identifier  in rows:
    ref_cgcs_valid[agr][prefix].add(identifier)
```

**Review by valearna — DISMISSED** [review]
Thanks @sweng66. I left a suggestion on how to improve the code with defaultdict

File: agr_literature_service/lit_processing/data_ingest/dqm_ingest/sort_dqm_json_reference_updates.py (hunk 1 above)
Comment (sweng66) [reply agr_literature_service/lit_processing/data_ingest/dqm_ingest/sort_dqm_json_reference_updates.py]: Wow! cool @valearna 

Just fixed to use defaultdict. Thanks.

**Review by valearna — APPROVED** [review]
Thanks @sweng66 !

**Pull request merged**